Source/WebKit/UIProcess/win/AutomationClientWin.h

```diff
@@ -40,14 +42,21 @@ namespace WebKit {
 class AutomationSessionClient final : public API::AutomationSessionClient {
 public:
     explicit AutomationSessionClient(const String&);
+    ~AutomationSessionClient();
 
     String sessionIdentifier() const override { return m_sessionIdentifier; }
 
     void requestNewPageWithOptions(WebKit::WebAutomationSession&, API::AutomationSessionBrowsingContextOptions, CompletionHandler<void(WebKit::WebPageProxy*)>&&) override;
     void didDisconnectFromRemote(WebKit::WebAutomationSession&) override;
 
+    void retainWebView(Ref<WebView>&&);
+    void releaseWebView(HWND);
+
 private:
     String m_sessionIdentifier;
+
+    static void close(WKPageRef, const void*);
+    HashMap<HWND, RefPtr<WebView>> m_webViews;
```

Comment:
Why does Windows port have to remember WebView instances? Other ports don't do that. The HWND will leak?

Reply:
Something must have gone wrong, but I forgot...
I'll try to disable this code and see what happens.

Reply:
I think Windows port is doing something wrong with lifetime of these objects because other ports don't do that.

Reply:
Maybe so, but I would like to fix it separately.

Reply:
I'm going to try that. Hold on please.

Reply:
I was wrong.
`requestNewPageWithOptions` calls `WebView::create` with null `parentWindow`.
It's a message-only window.
https://learn.microsoft.com/en-us/windows/win32/winmsg/window-features#message-only-windows
Then, the window of the view never gets closed.